pynecone/.templates/web/utils/state.js

```diff
@@ -132,19 +152,29 @@ export const updateState = async (
   setResult({ ...result, processing: true });
 
   // Pop the next event off the queue and apply it.
-  const event = state.events.shift();
-
+  const queue_event = state.events.shift();
   // Set new events to avoid reprocessing the same event.
   setState({ ...state, events: state.events });
 
   // Apply the event.
-  const eventSent = await applyEvent(event, router, socket);
-  if (!eventSent) {
-    // If no event was sent, set processing to false and return.
-    setResult({ ...state, processing: false });
+  if (queue_event.use_websocket == false) {
+
+    await applyRestEvent(queue_event, state, setResult)
+
+
   }
+  else {
+    const eventSent = await applyEvent(queue_event, router, socket);
+    if (!eventSent) {
+      // If no event was sent, set processing to false and return.
+      setResult({ ...state, processing: false });
```

Comment:
I think we should apply this `processing: false` flag for both types of events to keep all the logic for that flag in this function

Reply:
Yeah got it, I could probably do all the refactor in the follow up ticket #1012